### PR TITLE
rollout controller find sourceRevision

### PR DESCRIPTION
**Pull request by wangyikewxgm**
**What this PR does / why we need it**:

application use rollout as a trait, user shouldn't care sourceRevision.
when a new rollout restart, rollout controller use `status.LastTtargetRevision` as sourceRevision
application using rollout trait example is:
```
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: rollout-trait-test
  namespace: rollout-trait-test
spec:
  components:
    - name: express-server
      type: webservice
      properties:
        image: stefanprodan/podinfo:4.0.3
      traits:
        - type: rollout
          properties:
            targetSize: 5
```

**Which issue(s) this PR fixes**:

related issue:  #1605 

**Special notes for your reviewer**:

### CODE REVIEW

File: test/e2e-test/testdata/rollout/deployment/rolloutDefinition.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+apiVersion: core.oam.dev/v1beta1
+kind: TraitDefinition
+metadata:
+  name: rollout
```
Comment (wonderflow) [inline test/e2e-test/testdata/rollout/deployment/rolloutDefinition.yaml:4]: 这个做到 vela chart里面去，作为预装的trait，对应参数你再优化优化

File: test/e2e-test/testdata/rollout/deployment/webServiceDefinition.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Code generated by KubeVela templates. DO NOT EDIT.
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  name: webservice
```
Comment (wonderflow) [inline test/e2e-test/testdata/rollout/deployment/webServiceDefinition.yaml:5]: 用chart里预装那个，不要自己再创建了。

File: docs/examples/rollout-trait/webServiceDefinition.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Code generated by KubeVela templates. DO NOT EDIT.
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  name: webservice
```
Comment (wonderflow) [inline docs/examples/rollout-trait/webServiceDefinition.yaml:5]: 同前面的comment，不要 copy-paste这些已经存在的 Definition

Comment (wangyikewxgm) [reply docs/examples/rollout-trait/webServiceDefinition.yaml]: fixed

**Review by wonderflow — APPROVED** [review]
LGTM

**Pull request merged**